### PR TITLE
docs gen pass 2: local web font, add typescript hljs support

### PR DESCRIPTION
**Pull request by inoas**
- Replaces `OutputText` property `text` type of `String` with an enum of `OutputTypeData` which can be either `Binary` or `Text` and refactors all code to use that enum.
- Adds 3 woff2 web font files and thus removes the last CDN deps.
- Use system fallback fonts that github uses for body font and mono space text.
- Use local fonts if installed, this cuts down bandwidth for developers navigating between different gleam docs if they have Karla and Ubuntu Mono installed and/or allows them to switch the font locally.
- Adds typescript hljs support
- Edit: ~~Adds preload directives and moves tail loading assets to the head with defer attrib.~~ removed to make the PR smaller

### CODE REVIEW

**Comment by inoas** [comment]
The Rust code is probably neither elegant nor idiomatic or any good. But it compiles and the result is what I wanted to have in this PR.

**Comment by inoas** [comment]
Marked as draft as long as *docs generator pass 1* isn't merged.

**Comment by inoas** [comment]
This seems GTM from my side.

**Comment by inoas** [comment]
Anything left to do?

**Comment by inoas** [comment]
Also contains https://github.com/gleam-lang/gleam/pull/1821

**Comment by inoas** [comment]
Also added TypeScript support, as I forgot that last time around.

**Comment by inoas** [comment]
Is there anything I need to do?

**Comment by inoas** [comment]
Brought it up to date with main, removed changes unrelated to this PR, removed preloading improvements to shrink PR size, fixed the Changelog.

**Comment by inoas** [comment]
brought up to date with `main`/0.25.1 again.

**Comment by inoas** [comment]
Resolved conflicts with main again.

File: compiler-cli/src/fs.rs [hunk 1]
```diff
@@ -199,17 +199,17 @@ pub fn write_outputs_under(outputs: &[OutputFile], base: &Path) -> Result<(), Er
...
 pub fn write_output(file: &OutputFile) -> Result<(), Error> {
-    let OutputFile { path, text } = file;
-    write(path, text)
+    let OutputFile { path, data } = file;
+    write(path, data)
 }
 
-pub fn write(path: &Path, text: &str) -> Result<(), Error> {
-    write_bytes(path, text.as_bytes())
+pub fn write(path: &Path, data: &OutputFileData) -> Result<(), Error> {
```
Comment (lpil) [inline compiler-cli/src/fs.rs:211]: Could you revert the change to this function please?

File: compiler-core/src/io.rs [hunk 2]
```diff
@@ -50,9 +50,24 @@ pub trait Writer: std::io::Write + Utf8Writer {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum OutputFileData {
```
Comment (lpil) [inline compiler-core/src/io.rs:54]: How about we call this `Content`? Bit more concise

File: compiler-core/src/io.rs [hunk 3]
```diff
@@ -50,9 +50,24 @@ pub trait Writer: std::io::Write + Utf8Writer {
...
+    Binary(Vec<u8>),
+    Text(String),
+}
+
+impl OutputFileData {
+    pub fn to_bytes(&self) -> &[u8] {
```
Comment (lpil) [inline compiler-core/src/io.rs:60]: This should be called `as_bytes` as it's an inexpensive borrow

**Review by lpil — COMMENTED** [review]
Nice one!

File: compiler-cli/src/fs.rs (hunk 1 above)
Comment (inoas) [reply compiler-cli/src/fs.rs]: @lpil please take another look if it is fine now (just pushed).

**Review by lpil — COMMENTED** [review]
I think the merge went a bit wrong, there's a duplicate line in the changelog

**Comment by inoas** [comment]
> I think the merge went a bit wrong, there's a duplicate line in the changelog

Oh my, sorry about that. Fixed.

**Comment by inoas** [comment]
The failing tests seem unrelated?

**Review by lpil — APPROVED** [review]
Thank you!
